packages/flutter_tools/lib/src/version.dart

```diff
@@ -752,38 +757,50 @@ class GitTagVersion {
   }
 
   // TODO(fujino): Deprecate this https://github.com/flutter/flutter/issues/53850
-  /// Check for the release tag format pre-v1.17.0
+  /// Check for the release tag format of the form x.y.z-dev.m.n
```

Comment:
No longer `-dev`

Reply:
I had to support the `-dev` tag or else when I land this (before there are the new tags) the tool will be in a bad state (and pre-submit was failing because it couldn't parse the current version)

Reply:
It was either support the `-dev` tag, or change the `git describe --match` glob, and I figured if I support the `-dev` tag here, I can safely remove it after the next dev release with the new tag format.

Reply:
If possible, can you split out the -dev parsing code to more clearly delineate what logic can be removed after the next dev?

Reply:
This entire method can be removed in the next dev.